### ISSUE TITLE
decimal tests fail in nBallerina nightly

### ISSUE THREAD

**Issue by manuranga** [ballerina-platform/ballerina-lang#34748]
[Build](https://github.com/manuranga/nballerina/runs/4879369073?check_suite_focus=true) fails with

```
12-decimal/add1-v.bal failed
12-decimal/const7-v.bal failed
12-decimal/rem1-v.bal failed
```

**Comment by KavinduZoysa** [comment]
```ballerina
public function main() {
    io:println(1E-6143d + -1E-6143d); // @output 0E-6143
}
```
If the result of the decimal operation is zero with trailing zeros, we print that value as it is.
But it should be only zero.
Above test failures happen under the const folding of nballerina because constant operations are evaluated using jballerina.

According to spec we implement `ANSI X3.274 subset of IEEE 754-2008` http://speleotrove.com/decimal/dax3274.html and it says,
```
If the coefficient in a result has the value zero, then the sign is set to 0 and (unless the operation is quantize) the exponent is set to 0.
```

Therefore we print use trailing zeros if the value is zero.

**Comment by KavinduZoysa** [comment]
Fixed in #34755 and verified with nballerina
